core/code/chat.js

```diff
@@ -521,7 +533,31 @@ window.chat.renderMarkup = function (markup) {
   return msg;
 };
 
-window.chat.renderTimeCell = function(time, classNames) {
+function transformMessage(markup) {
```

Comment:
If we want to transform data, we should do it as close to server as possible - entity_something.js

Imho we should update the rendering instead.

Reply:
chat data is completly processed in chat.js
`renderMarkup` is used to process new received data and store it in a local cache.

so here I changed the server data right after it is received and as close as possible to the point were it is rendered. (not displayed)

Other possible locations for a "transformMessage" call would be: 'renderMsgRow', 'parseMsgData'

Reply:
> Other possible locations for a "transformMessage" call would be: 'renderMsgRow', 'parseMsgData'

I prefer in `renderMsgRow`, exposing the function as `window.chat.transformMarkup`, something like
```js
var markup = window.chat.transformMarkup(data.markup);
var msg = window.chat.renderMarkup(markup);
...
// or
var markup = window.chat.transformMarkup(data); // give parsed data as well
var msg = window.chat.renderMarkup(markup);
```

Reply:
Wouldn't it be cool if IITC would do the whole text parsing? 
And plugins could rely on always receiving the same format?

Let say nia changes the text from "destroyed" to "eliminated". Wouldn't it be cool if it just requires an IITC update instead of patching 10 plugins?  --- okay, if we can agree that IITC updated shouldn't take a year.

---
'exposing' .. I prefer private data, private functions and strict interfaces. Exposed functions and data entice to manipulate them. Making code harder to change or leads to unpredicted results.

Reply:
plugins based on chat data use raw data directly via the chat hooks, they don't use the one parsed by the core plugin as far as I know
also, it's about rendering, the markup is not the place to check and rewrite what the data is about, it's better to leave it in a specific places.
Then, you may have a plugin that see : this message is a "destroyed field" message, about this "agent" on this "portal", and then create a markup for it from scratch.

Reply:
> raw data 

not "result" ? atleast all stock plugins use "result" instead of "raw"

> EDIT: about extracting data, that was one of my objective with the comm-filter plugin for instance

isn't this excatly what I tried to say? Why doing this in every plugin when IITC can standarized it?

Reply:
yeah, sure, the 'transformMessage' function doesn't fit well in the chat.js code style.

Reply:
but in the end I don't care.
This is just a feature request with code. Do what ever you like with it.
I don't wanne wast our time with endless discussions.

Reply:
> not "result" ? atleast all stock plugins use "result" instead of "raw"

`.result == .raw.result` is what the intel is sending, unprocessed

> isn't this excatly what I tried to say? Why doing this in every plugin when IITC can standarized it?

I was agreeing to your point, but stating that markup isn't the place for it